### DISCUSSION TITLE
How would you use an API with Financial Freedom?

### DISCUSSION THREAD

**Discussion by danpastori** [serversideup/financial-freedom#117] [Project Discussions]
I'm a huge fan of APIs for all apps. I was wondering how you'd use an API with Financial Freedom? Our goal is to have Financial Freedom self hostable and easy to set up on your own machine. Would you use an API to import transactions from out side of the app? Would you want to dump your transaction data into other platforms?

Just curious so I can add some endpoints and gauge a use case.

**Comment by siegenthalerroger** [comment]
Definitely more interested in an API to import transactions in to the app, however exports are a big deal aswell obviously.

**Reply by danpastori** [reply]
Awesome! The transactions API will be the most feature rich. I have a ton of filters to get the data you need and easy to create and import through the API.

**Comment by PrplHaz4** [comment]
Importing and exporting transactions as stated above are by far the most important.  We're here to avoid lock-in.

Another use would be enriching/categorizing already imported transactions (with Amazon order data for instance)...

**Comment by GeekDougle-Noya** [comment]
I'm going to be getting into self hosted home automation and I would love to have a dashboard of spend and available budget available for the family on a smart display.  I envision having the ability to designate what categories I want available to query and the display just gets the available budget for the allowed categories.

I could also see notifications for spend over a certain threshold being tied to different tools (ifttt, slack, or similar)

**Comment by hackschnitzel09** [comment]
Bunq as a bank has an API as well so keeping the account with ff in sync could be an idea
